tests/AutoReview/ProjectCodeTest.php

```diff
@@ -299,6 +315,52 @@ function ($item) {
         );
     }
 
+    public function provideDataProviderMethodNameCases()
+    {
+        $data = array();
+
+        $testClassNames = $this->getTestClasses();
+
+        foreach ($testClassNames as $testClassName) {
+            $reflection = new \ReflectionClass($testClassName);
+
+            if (!$reflection->isSubclassOf('PHPUnit\Framework\TestCase')) {
+                continue;
+            }
+
+            $dataProviderMethodNames = array();
+
+            foreach ($reflection->getMethods(\ReflectionMethod::IS_PUBLIC) as $method) {
+                if (false === $method->getDocComment()) {
+                    continue;
+                }
+
+                $docBlock = new DocBlock($method->getDocComment());
+
+                $dataProviderAnnotations = $docBlock->getAnnotationsOfType('dataProvider');
+
+                foreach ($dataProviderAnnotations as $dataProviderAnnotation) {
+                    if (false === preg_match('/@dataProvider\s+(?P<methodName>\w+)/', $dataProviderAnnotation->getContent(), $matches)) {
```

Comment:
`preg_match()` only returns `false` when an error occurs but the pattern looks good, did you mean `0`? Otherwise, what kind of error do you expect here?

Reply:
Hehe, will fix tomorrow!